pkg/loadbalancerservice/router.go

```diff
@@ -0,0 +1,135 @@
+package loadbalancerservice
+
+import (
+	"slices"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultRouterServiceName            = "router-default"
+	defaultRouterServiceNamespace       = "openshift-ingress"
+	defaultRouterServiceAnnotationKey   = "ingresscontroller.operator.openshift.io/owning-ingresscontroller"
+	defaultRouterServiceAnnotationValue = "default"
+)
+
+type serviceUpdateFunction func([]string) error
+
+func defaultRouterWatch(ipAddresses, nicNames []string, updateFunc serviceUpdateFunction, stopCh <-chan struct{}) {
+	for {
+		ips, err := defaultRouterListenAddresses(ipAddresses, nicNames)
+		if err != nil {
+			klog.Errorf("unable to determine default router listening addresses: %v", err)
+			continue
+		}
+		if err := updateFunc(ips); err != nil {
+			klog.Errorf("unable to update default router service status: %v", err)
+			continue
+		}
+		break
+	}
+
+	updateChan := make(chan netlink.AddrUpdate)
+	doneChan := make(chan struct{})
+	err := netlink.AddrSubscribe(updateChan, doneChan)
+	if err != nil {
+		klog.Errorf("unable to subscribe to IP address changes: %v", err)
+		return
+	}
+
+	for {
+		select {
+		case <-updateChan:
+			ips, err := defaultRouterListenAddresses(ipAddresses, nicNames)
+			if err != nil {
+				klog.Errorf("unable to determine default router listening addresses: %v", err)
+				break
+			}
+			err = updateFunc(ips)
+			klog.Infof("P333. This is something I am waiting for: %v", err)
+			if err != nil {
+				klog.Errorf("unable to update default router service status: %v", err)
+				break
+			}
+		case <-stopCh:
+			klog.Info("default router watcher stopping")
+			return
+		}
+	}
+}
+
+func isDefaultRouterService(svc *corev1.Service) bool {
+	annotationValue, annotationFound := svc.Labels[defaultRouterServiceAnnotationKey]
+	return annotationFound &&
+		annotationValue == defaultRouterServiceAnnotationValue &&
+		svc.Name == defaultRouterServiceName &&
+		svc.Namespace == defaultRouterServiceNamespace
+}
+
+func defaultRouterListenAddresses(ipAddresses, nicNames []string) ([]string, error) {
+	allowedAddresses, err := config.AllowedListeningIPAddresses()
+	if err != nil {
+		return nil, err
+	}
+
+	allowedNicNames, err := config.AllowedNICNames()
+	if err != nil {
+		return nil, err
+	}
+
+	ipList := make([]string, 0)
```

Comment:
what do you think about this?
```suggestion
	ipList := make([]string, 0, len(allowedNicNames) + len(nicNames)*2)
```